src/olympia/api/tests/test_fields.py

```diff
@@ -0,0 +1,264 @@
+# -*- coding: utf-8 -*-
+from django.core.exceptions import ValidationError
+
+from mock import Mock
+from nose.tools import eq_
+from rest_framework.request import Request
+from rest_framework.serializers import Serializer
+from rest_framework.test import APIRequestFactory
+
+from olympia.addons.models import Addon
+from olympia.api.fields import (ESTranslationSerializerField,
+                                TranslationSerializerField)
+from olympia.amo.tests import TestCase
+from olympia.translations.models import Translation
+
+
+class _TestTranslationSerializerField(object):
+    """
+    Base parent class for translation fields tests.
+
+    This allows the same tests to be applied to TranslationSerializerField
+    (MySQL) and ESTranslationSerializerField (ElasticSearch) since everything
+    should work transparently.
+    """
+    field_class = TranslationSerializerField
+
+    def setUp(self):
+        super(_TestTranslationSerializerField, self).setUp()
+        self.factory = APIRequestFactory()
+        self.addon = Addon.objects.get(pk=3615)
+        Translation.objects.create(id=self.addon.name.id, locale='es',
+                                   localized_string=u'Name in Español')
+
+    def _test_expected_dict(self, field):
+        result = field.field_to_native(self.addon, 'name')
+        expected = {
+            'en-US': unicode(Translation.objects.get(id=self.addon.name.id,
+                                                     locale='en-US')),
+            'es': unicode(Translation.objects.get(id=self.addon.name.id,
+                                                  locale='es')),
+        }
+        eq_(result, expected)
+
+        result = field.field_to_native(self.addon, 'description')
+        expected = {
+            'en-US': Translation.objects.get(id=self.addon.description.id,
+                                             locale='en-US'),
+        }
+        eq_(result, expected)
+
+    def _test_expected_single_string(self, field):
+        result = field.field_to_native(self.addon, 'name')
+        expected = unicode(self.addon.name)
+        eq_(result, expected)
+
+        result = field.field_to_native(self.addon, 'description')
+        expected = unicode(self.addon.description)
+        eq_(result, expected)
+
+    def test_from_native(self):
+        data = u'Translatiön'
+        field = self.field_class()
+        result = field.from_native(data)
+        eq_(result, data)
+
+        data = {
+            'fr': u'Non mais Allô quoi !',
+            'en-US': u'No But Hello what!'
+        }
+        field = self.field_class()
+        result = field.from_native(data)
+        eq_(result, data)
+
+        data = ['Bad Data']
+        field = self.field_class()
+        result = field.from_native(data)
+        eq_(result, unicode(data))
+
```

Comment:
I see, if we don't need it, maybe better remove it rather than keeping it around. Otherwise I think we don't need to care about old clients here at all.